artifactory/utils/repositoryutils.go

```diff
@@ -98,3 +69,60 @@ func IsRemoteRepo(repoName string, serviceManager artifactory.ArtifactoryService
 	}
 	return repoDetails.GetRepoType() == "remote", nil
 }
+
+// GetFilteredRepositories returns the names of all repositories filtered by their names and types.
+// includePatterns - patterns of repository names (can contain wildcards) to include in the results. A repository's name
+// must match at least one of these patterns in order to be included in the results. If includePatterns' length is zero,
+// all repositories are included.
+// excludePatterns - patterns of repository names (can contain wildcards) to exclude from the results. A repository's name
+// must NOT match any of these patterns in order to be included in the results.
+// repoType - only repositories of this type will be returned. Pass nil to return repositories of all types.
+func GetFilteredRepositories(servicesManager artifactory.ArtifactoryServicesManager, includePatterns, excludePatterns []string, repoType *RepoType) ([]string, error) {
+	filterParams := services.RepositoriesFilterParams{RepoType: repoType.String()}
+	repos, err := servicesManager.GetAllRepositoriesFiltered(filterParams)
+	if err != nil {
+		return nil, err
+	}
+	return filterRepositoryNames(repos, includePatterns, excludePatterns)
+}
+
+func filterRepositoryNames(repos *[]services.RepositoryDetails, includePatterns, excludePatterns []string) ([]string, error) {
```

Comment:
I think that the logic in thsi function can optimized as follows.
If no include patterns were sent, we can avoid running `filepath.Match` on the list of repositories. The same goes for an empty exclude patterns list. This is important because this API is expected to be used with hundreds or even thousands of repositories, and the `filepath.Match` isn't trivial in terms of cost.